### ISSUE TITLE
CiviCRM 5.12.0 issue:  Group not Set for sync

### ISSUE THREAD

**Issue by bassngom** [veda-consulting/uk.co.vedaconsulting.outlookapi#65]
Hi, 
In  CiviCRM 5.12.0, I'm trying to use OutlookForCiviCRM plugin and the civicrm extension to synchronize contacts and groups between outlook and Civi. 
However, I get the the following issue.

![image](https://user-images.githubusercontent.com/49524899/58707997-07ca0280-83a6-11e9-95ce-84cfaec9510a.png)

I have installed the 3.0.7 beta version of OutlookForCiviCRM and 

![unnamed](https://user-images.githubusercontent.com/49524899/58708018-1b756900-83a6-11e9-9556-9219a8da4da9.png)

I have also checked the Sync to outlook box in the group settings 

![image (1)](https://user-images.githubusercontent.com/49524899/58708035-23cda400-83a6-11e9-816f-2c89d417a7f8.png)

and the group contains also some members.

Please can you help me to solve this problem? I want also to synchronize all contacts from Civi, how can I do this?

Cheers
Bass

**Comment by jusfreeman** [comment]
Just noting that I have seen this problem too and thought initially #61 would fix it - unfortunately not.

Agileware Ref: CIVIOLK-1

**Comment by bassngom** [comment]
> Just noting that I have seen this problem too and thought initially #61 would fix it - unfortunately not.
> 
> We're working / looking for a fix.
> 
> Agileware Ref: CIVIOLK-1

We will appreciate that you share your experience with us

**Comment by jusfreeman** [comment]
> We will appreciate that you share your experience with us

@bassngom yes, when we have the fix, will share it (no need to be confused)

**Comment by agileware-pengyi** [comment]
@bassngom Please check if #66 help you out or not.

**Comment by bassngom** [comment]
@agileware-pengyi I try it but that does not resolve myproblem. Should I disconnect the civiForOutlook plugin after reporting these fix?

**Comment by agileware-pengyi** [comment]
@bassngom No, you just need to apply the PR to the extension. I tested on the plugin version 3.0.4. Another thing you can check is the api response. See if the api `outlook` with action `getgroupcontacts` give you a correctly formatted response.

**Comment by AwaMakoa** [comment]
@agileware-pengyi I have the same problem as bass ngom
can you help me please ?

**Comment by samuelsov** [comment]
I've tried to debug the site mentioned by @AwaMakoa and after applying PR #66, I've ensured that the api call was sending the expected information. The REST test with the same credential as those given in Outlook is working fine:
```
https://mysite.net/vendor/civicrm/civicrm-core/extern/rest.php?entity=CiviOutlook&action=getgroupcontacts&api_key=XXX&key=YYY&json=sequential:1 
```

Now, if we do the sync in Outlook with the api sending 2+ contacts, it gives the error "Group not set for sync" but if we tweak the api to send only one contact, it does work partially. The contact is updated although the group doesn't seems to be in every case.

To give some real data, this json does not give the error in Outlook and update the contact in Outlook:
![Screenshot_2021-02-16 https civicrm-rgpd makoa net(1)](https://user-images.githubusercontent.com/372004/108103694-4ad77800-7058-11eb-9139-394146c18191.png)

Howerver, this gives the error  "Group not set for sync" :
![Screenshot_2021-02-16 https civicrm-rgpd makoa net](https://user-images.githubusercontent.com/372004/108103681-4612c400-7058-11eb-8317-09a3c5c948ac.png)

Anyone with the same problem in a recent CiviCRM installation (5.29+) and PR #66 (and #63) applied ? 

Without the code for the msi file, I don't know what else to do to help resolving this.

**Comment by aeszq** [comment]
Hi, I'm just wondering whether there is a fix for the issue bassngom had raised? I'm having the same issue.

**Comment by AwaMakoa** [comment]
Hi,
no still no solution
